http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpSamDAO.scala

```diff
@@ -256,6 +256,46 @@ class HttpSamDAO[F[_]](httpClient: Client[F],
         }
     } yield ()
 
+  def createResourceWithWorkspaceParent[R](resource: R,
+                                           creatorEmail: WorkbenchEmail,
+                                           userInfo: UserInfo,
+                                           workspaceId: WorkspaceId
+  )(implicit
+    sr: SamResource[R],
+    encoder: Encoder[R],
+    ev: Ask[F, TraceId]
+  ): F[Unit] =
+    for {
+      traceId <- ev.ask
+      // TODO: Check if this is the best way to get the token, there are no petSA for Workspace resources, only google-projects
+      authHeader = Authorization(Credentials.Token(AuthScheme.Bearer, userInfo.accessToken.token))
+      loggingCtx = Map("traceId" -> traceId.asString)
+      _ <- logger.info(loggingCtx)(
+        s"Creating ${sr.resourceType.asString} resource in sam v2 for ${workspaceId}/${sr.resourceIdAsString(resource)}"
+      )
+      _ <- metrics.incrementCounter(s"sam/createResource/${sr.resourceType.asString}")
+      policies = Map[SamPolicyName, SamPolicyData](
+        SamPolicyName.Creator -> SamPolicyData(List(creatorEmail), List(SamRole.Creator))
+      )
+      parent = SerializableSamResource(SamResourceType.Workspace, WorkspaceResourceSamResourceId(workspaceId))
```

Comment:
Cool - so this method + logic works for both `kubernetes-app` and `kubernetes-app-shared` resource types, correct?

Reply:
Yep!